// ant-design-vue/lib/vc-tree-select/src/PropTypes.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectPropTypes = undefined;

var _vueTypes = require('../../_util/vue-types');

var _vueTypes2 = _interopRequireDefault(_vueTypes);

var _strategies = require('./strategies');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var SelectPropTypes = exports.SelectPropTypes = {
  // className: PropTypes.string,
  prefixCls: _vueTypes2['default'].string,
  multiple: _vueTypes2['default'].bool,
  filterTreeNode: _vueTypes2['default'].any,
  showSearch: _vueTypes2['default'].bool,
  disabled: _vueTypes2['default'].bool,
  showArrow: _vueTypes2['default'].bool,
  allowClear: _vueTypes2['default'].bool,
  defaultOpen: _vueTypes2['default'].bool,
  open: _vueTypes2['default'].bool,
  transitionName: _vueTypes2['default'].string,
  animation: _vueTypes2['default'].string,
  choiceTransitionName: _vueTypes2['default'].string,
  // onClick: PropTypes.func,
  // onChange: PropTypes.func,
  // onSelect: PropTypes.func,
  // onDeselect: PropTypes.func,
  // onSearch: PropTypes.func,
  searchPlaceholder: _vueTypes2['default'].string,
  placeholder: _vueTypes2['default'].any,
  inputValue: _vueTypes2['default'].any,
  value: _vueTypes2['default'].any,
  defaultValue: _vueTypes2['default'].any,
  label: _vueTypes2['default'].any, // vnode
  defaultLabel: _vueTypes2['default'].any,
  labelInValue: _vueTypes2['default'].bool,
  dropdownClassName: _vueTypes2['default'].string,
  dropdownStyle: _vueTypes2['default'].object,
  dropdownPopupAlign: _vueTypes2['default'].object,
  dropdownVisibleChange: _vueTypes2['default'].func,
  maxTagTextLength: _vueTypes2['default'].number,
  showCheckedStrategy: _vueTypes2['default'].oneOf([_strategies.SHOW_ALL, _strategies.SHOW_PARENT, _strategies.SHOW_CHILD]),
  treeCheckStrictly: _vueTypes2['default'].bool,
  treeIcon: _vueTypes2['default'].bool,
  treeLine: _vueTypes2['default'].bool,
  treeDefaultExpandAll: _vueTypes2['default'].bool,
  treeDefaultExpandedKeys: _vueTypes2['default'].arrayOf(String),
  treeCheckable: _vueTypes2['default'].any, // bool vnode
  treeNodeLabelProp: _vueTypes2['default'].string,
  treeNodeFilterProp: _vueTypes2['default'].string,
  treeData: _vueTypes2['default'].array,
  treeDataSimpleMode: _vueTypes2['default'].oneOfType([_vueTypes2['default'].bool, _vueTypes2['default'].object]),
  loadData: _vueTypes2['default'].func,
  dropdownMatchSelectWidth: _vueTypes2['default'].bool,
  notFoundContent: _vueTypes2['default'].any,
  children: _vueTypes2['default'].any,
  autoFocus: _vueTypes2['default'].bool,
  getPopupContainer: _vueTypes2['default'].func,
  switcherIcon: _vueTypes2['default'].func,
  inputIcon: _vueTypes2['default'].any,
  removeIcon: _vueTypes2['default'].any,
  clearIcon: _vueTypes2['default'].any
};
